### PR TITLE
Upgrade nightly used in CI

### PR DESCRIPTION
**Pull request by jdisanti**
_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Comment by jdisanti** [comment]
Looks like the CI failure is caused by new rustdoc warnings, which, paired with `-D warnings`, is causing cargo-check-external-types to fail.

File: aws/rust-runtime/aws-config/src/default_provider/retry_config.rs [hunk 1]
```diff
@@ -81,7 +81,7 @@ impl Builder {
         self
     }
 
-    /// Attempt to create a [RetryConfig](aws_smithy_types::retry::RetryConfig) from following sources in order:
+    /// Attempt to create a [RetryConfig] from following sources in order:
```
Comment (ysaito1001) [inline aws/rust-runtime/aws-config/src/default_provider/retry_config.rs:84]:
```suggestion
    /// Attempt to create a [`RetryConfig`] from following sources in order:
```

**Review by ysaito1001 — APPROVED** [review]
Thanks for fixing a lot of doc links!

**Comment by jdisanti** [comment]
It looks like https://github.com/smithy-lang/smithy-rs/pull/3298 was unintentionally merged into this PR before it was merged to main. So this PR included the MSRV upgrade as well.

**Pull request merged**